include/zephyr/bluetooth/hci_vs.h

```diff
@@ -216,9 +216,29 @@ struct bt_hci_evt_vs {
 } __packed;
 
 #define BT_HCI_EVT_VS_FATAL_ERROR              0x02
+
+#define BT_HCI_EVT_VS_ERROR_TYPE_SYSTEM        0x01
+#define BT_HCI_EVT_VS_ERROR_TYPE_CTRL_ASSERT   0x02
+struct bt_hci_vs_fata_error_cpu_data_cortex_m {
+	uint32_t a1;
+	uint32_t a2;
+	uint32_t a3;
+	uint32_t a4;
+	uint32_t ip;
+	uint32_t lr;
+	uint32_t xpsr;
+} __packed;
+
+#define BT_HCI_EVT_VS_ERROR_CPU_TYPE_CORTEX_M  0x01
+struct bt_hci_vs_fatal_error_stack_frame {
+	uint32_t reason;
+	uint8_t cpu_type;
+	uint8_t cpu_data[0];
+} __packed;
+
 struct bt_hci_evt_vs_fatal_error {
-	uint64_t pc;
-	uint8_t  err_info[0];
+	uint8_t type;
+	uint8_t data[0];
```

Comment:
I realize that we haven't done much documentation in this file, but it is a public header file that are occasionally used by applications. Should we consider documenting this a bit? At least I don't think it's clear to everyone later that the `data` can be a `struct bt_hci_vs_fatal_error_stack_frame` and that `cpu_data` can be a `struct bt_hci_vs_fata_error_cpu_data_cortex_m`.

Reply:
For sure I'll add more extensive documentation. I didn't want to spend time on that in case the whole approach to the topic is not accepted.